src/java/org/apache/cassandra/tools/BootstrapMonitor.java

```diff
@@ -34,6 +34,7 @@ public class BootstrapMonitor extends JMXNotificationProgressListener
     private final SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss,SSS");
     private final PrintStream out;
     private final Condition condition = newOneTimeCondition();
+    private Exception error;
```

Comment:
I think this is going to need to be `volatile`, as it's written by one thread and accessed by another.